x-pack/plugins/maps/server/data_indexing/indexing_routes.ts

```diff
@@ -104,6 +104,40 @@ export function initIndexingRoutes({
     }
   );
 
+  router.delete(
+    {
+      path: `${INDEX_FEATURE_PATH}/{featureId}`,
+      validate: {
+        params: schema.object({
+          featureId: schema.string(),
+        }),
+        body: schema.object({
+          index: schema.string(),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      try {
+        const { body: resp } = await context.core.elasticsearch.client.asCurrentUser.delete({
+          index: request.body.index,
+          id: request.params.featureId,
+          refresh: true,
+        });
+        if (resp.result === 'Error') {
+          throw resp;
+        } else {
+          return response.ok({ body: { success: true } });
+        }
+      } catch (error) {
+        logger.error(error);
+        return response.custom({
+          body: error.message,
+          statusCode: 500,
```

Comment:
not sure 500 is the right error code. 500 means something with the service is wrong. Failure coulde be lots of reasons ranging from permissions to something is wrong with elasticsearch.

Reply:
It's catch-all-ish for sure. The most likely error is that the feature doesn't exist. I've updated it to a 404

Reply:
404 is resource not found and is also not a good catch all status code. 404 is the right code in the case delete failed because the document with _id could not be found.

If the user is not authorized, then 403 should be returned.